clang/include/clang/Sema/Sema.h

```diff
@@ -13068,21 +13068,23 @@ void Sema::addIntelSingleArgAttr(Decl *D, const AttributeCommonInfo &CI,
       return;
     E = ICE.get();
     int32_t ArgInt = ArgVal.getSExtValue();
-    if (CI.getParsedKind() == ParsedAttr::AT_SYCLIntelNumSimdWorkItems ||
-        CI.getParsedKind() == ParsedAttr::AT_IntelReqdSubGroupSize ||
+    if (CI.getParsedKind() == ParsedAttr::AT_IntelReqdSubGroupSize ||
         CI.getParsedKind() == ParsedAttr::AT_IntelFPGAMaxReplicates) {
       if (ArgInt <= 0) {
         Diag(E->getExprLoc(), diag::err_attribute_requires_positive_integer)
             << CI << /*positive*/ 0;
         return;
       }
     }
-    if (CI.getParsedKind() == ParsedAttr::AT_SYCLIntelMaxGlobalWorkDim) {
+    if (CI.getParsedKind() == ParsedAttr::AT_SYCLIntelMaxGlobalWorkDim ||
+        CI.getParsedKind() == ParsedAttr::AT_SYCLIntelNumSimdWorkItems) {
```

Comment:
If NumSimdWorkItems needs to divide WorkGroupSize, it cannot be 0 right- correct

Reply:
I am handling this inside handleNumSimd() when the value is 0.

Reply:
I kept this here to give the diagnostic for negative value for template instantiation. Here we only care about  value dependency. since we are not here  !E->isInstantiationDependent()) 

// expected-error@+1{{'num_simd_work_items' attribute requires a non-negative integral compile time constant expression}}
[[intel::num_simd_work_items(N)]] void func3() {}

int check() {
  
  //expected-note@+1{{in instantiation of function template specialization 'func3<-1>' requested here}}
  func3<-1>();
  return 0;
}

Reply:
I kept this check here to give diagnostic for negative value during template instantiation. 

I did not move this check since we will refactor this attribute later that will remove it from here 
// expected-error@+1{{'num_simd_work_items' attribute requires a non-negative integral compile time constant expression}}
[[intel::num_simd_work_items(N)]] void func3() {}

int check() {
  
  //expected-note@+1{{in instantiation of function template specialization 'func3<-1>' requested here}}
  func3<-1>();
  return 0;
}

Reply:
Ok. Thanks for clarification